src/plugins/embeddable/public/lib/actions/edit_panel_action.ts

```diff
@@ -35,11 +36,18 @@ export class EditPanelAction implements Action<ActionContext> {
   public readonly type = ACTION_EDIT_PANEL;
   public readonly id = ACTION_EDIT_PANEL;
   public order = 50;
+  public currentAppId: string | undefined;
 
   constructor(
     private readonly getEmbeddableFactory: EmbeddableStart['getEmbeddableFactory'],
     private readonly application: ApplicationStart
-  ) {}
+  ) {
+    if (this.application?.currentAppId$) {
+      this.application.currentAppId$
+        .pipe(take(1))
+        .subscribe((appId: string | undefined) => (this.currentAppId = appId));
```

Comment:
I've created a discuss issue to improve this syntax: https://github.com/elastic/kibana/issues/65224

Reply:
This is a great issue to bring up. I did switch currentAppId$ to a behaviorSubject at some point in this PR, but most tests aren't set up to ignore the first value and were failing, so I figured it would be better to do it in a separate PR.